core/src/main/scala/com/mesosphere/usi/core/SchedulerAdapter.scala

```diff
@@ -0,0 +1,199 @@
+package com.mesosphere.usi.core
+
+
+import akka.{Done, NotUsed}
+import akka.stream.{KillSwitches, Materializer, OverflowStrategy}
+import akka.stream.scaladsl.{Flow, Sink, SinkQueueWithCancel, Source, SourceQueueWithComplete}
+import com.mesosphere.usi.core.Scheduler.{SpecInput, StateOutput}
+import com.mesosphere.usi.core.models.{SpecEvent, SpecUpdated, SpecsSnapshot, StateEvent, StateSnapshot}
+
+import scala.concurrent.{Future, Promise}
+
+
+/**
+  * SchedulerAdapter provides a set of simple interfaces to the scheduler flow.
+  * @param schedulerFlow
+  * @param mat
+  */
+class SchedulerAdapter(schedulerFlow: Flow[SpecInput, StateOutput, NotUsed])(implicit mat: Materializer) {
+
+  private val extractSnapshot: Flow[SpecEvent, (SpecsSnapshot, Source[SpecUpdated, Any]), NotUsed] = Flow[SpecEvent]
+    .prefixAndTail(1) // First event must be a snashot, let's take it
+    .map {
+      case (Seq(specsSnapshot: SpecsSnapshot), source) =>
+        val specUpdatedSource = source.map {
+          case specUpdated: SpecUpdated => specUpdated
+          case other => throw new IllegalArgumentException(s"expected SpecUpdated but got $other")
+        }
+        (specsSnapshot, specUpdatedSource)
+      case _ =>
+        throw new IllegalArgumentException("scheduler spec stream must start with a snapshot")
+    }
+
+  private val mergeSnapshot: Flow[(StateSnapshot, Source[StateEvent, Any]), StateEvent, NotUsed] = Flow[StateOutput]
+    .flatMapConcat {
+      case (snapshot, updates) =>
+        updates.prepend(Source.single(snapshot))
+    }
+
+
+  /**
+    * Represents the scheduler as a Sink and Source.
+    *
+    * This method will materialize the scheduler first, then Sink and Source can be materialized independently.
+    * Please note that the first spec event coming to the Sink must be a SpecsSnapshot otherwise the stream will fail.
+    *
+    * @return Source which produces StateEvents and Sink which accepts SpecEvents
+    */
+  def asSourceAndSink(): (Source[StateEvent, NotUsed], Sink[SpecEvent, NotUsed]) = {
```

Comment:
What do you think about passing the `SpecsSnapshot`? Or rather why is it not passed here?

Reply:
Check the method below :)

Reply:
See my comment below ;)

Reply:
I'd much rather have it in the docs or are we lifting the hello world examples to docs?